src/spec.c

```diff
@@ -178,28 +214,27 @@ IndexSpec *IndexSpec_CreateNew(RedisModuleCtx *ctx, RedisModuleString **argv, in
 
   CursorList_AddSpec(&RSCursors, sp->name, RSCURSORS_DEFAULT_CAPACITY);
 
-  if (sp->flags & Index_Temporary) {
```

Comment:
Why not setting the timer for a temporary index?

Reply:
We do, only the order of operation was changed, so the counter starts to tick just before starting the scan:
```
  // Create the indexer
  sp->indexer = NewIndexer(sp);
  if (IndexSpec_OnCreate) {
    IndexSpec_OnCreate(sp);
  }

  if (sp->flags & Index_Temporary) {
    IndexSpec_SetTimeoutTimer(sp);
  }

  IndexSpec_ScanAndReindex(sp);
```

Reply:
Sorry I see you do set the timer